src/Xeno/Types.hs

```diff
@@ -7,12 +7,13 @@ module Xeno.Types where
 
 import Control.DeepSeq
 import Control.Exception
+import Data.ByteString (ByteString)
 import Data.Typeable
 import GHC.Generics
 
 data XenoException
   = XenoStringIndexProblem
-  | XenoParseError
+  | XenoParseError ByteString
```

Comment:
@chrisdone My intuition was that an error is thrown only once, so it doesn't really matter if it is slow to throw an exception. This might be a bit naive though. Do the exceptions affect the non-error parts of the code negatively when it comes to performance?